src/client/particles.cpp

```diff
@@ -680,22 +680,22 @@ void ParticleManager::stepParticles(float dtime)
 {
 	MutexAutoLock lock(m_particle_list_lock);
 
-	for (auto it = m_particles.begin(); it != m_particles.end();) {
-		std::unique_ptr<Particle> &p = *it;
-		if (p->isExpired()) {
-			ParticleSpawner *parent = p->getParent();
+	for (size_t i = 0; i < m_particles.size();) {
+		Particle &p = *m_particles[i];
+		if (p.isExpired()) {
```

Comment:
alternatively not sure if a list would be good

Reply:
A linked list of all particles? Meh.